### PR TITLE
workflows: check MSRV in CI.

### PR DESCRIPTION
**Pull request by kivikakk**
* [x] Run CI against the MSRV.
  * Ensure the naming means we don't have to adjust the branch protection rules when we adjust the MSRV.
* [x] ~~Update the MSRV used in CI when it's updated in `Cargo.toml`.~~ There's no clear time to do such an update; CI can instead check this for us.

### CODE REVIEW

**Comment by kivikakk** [comment]
Hooboy. This is frustrating enough that I want to pull the CLI binary out as its own crate — which makes sense, given I want different MSRV disciplines between them.

**Comment by digitalmoksha** [comment]
> frustrating enough that I want to pull the CLI binary out as its own crate

I understand - I'm at 1.73 and while starting a new branch I had to up level because of the CLI.

However it would be a bit of a bummer, as it's nice to be able to debug directly using the RustRover IDE - it runs the CLI and I can just step through code.

Also, I think your compliance specs are run using the CLI.

Can we not pin the CLI crates to the version that we're supporting? Is some critical functionality/bug fixes getting lost?

**Comment by kivikakk** [comment]
> Also, I think your compliance specs are run using the CLI.

Yep, my current WIP builds the library on nightly/beta/stable+MSRV and runs unit tests with it, then builds the binary on nighty/beta/stable and runs specs with it.

> Can we not pin the CLI crates to the version that we're supporting? Is some critical functionality/bug fixes getting lost?

There's just a bunch of crates that are starting to get rather old — they left 1.62 behind a while ago — and so as a matter of course I don't really love leaving them back. This mostly applies to the CLI ones. The recently added `in-place` is hard to support too, though I can just as easily tear that out again, it's minor.

**Comment by kivikakk** [comment]
I think it's easier just to set a standard MSRV across both, so yeah, I'm going to do the work to get us back to 1.62.1 👍

**Comment by digitalmoksha** [comment]
If there are say security or performance fixes in a crate, when a user builds the library they should be able to specify a more recent version of that crate, right?  My Rust dependency knowledge is still young.

**Comment by kivikakk** [comment]
> when a user builds the library they should be able to specify a more recent version of that crate, right? My Rust dependency knowledge is still young.

Mine too, lol 😅

I'm not quite sure — probably? If it gets to that point, I'm sure someone will let us know!

**Comment by kivikakk** [comment]
lol okay. I was thinking `cargo update` would know not to update to versions out of its MSRV. It doesn't, [yet](https://github.com/rust-lang/cargo/issues/9930), but it wouldn't help us anyway until we bump our MSRV to one with that Cargo anyway, I guess. Yeesh.

**Comment by kivikakk** [comment]
Happy to take feedback from anyone interested; will merge in a day or two otherwise!

**Comment by digitalmoksha** [comment]
This looks great @kivikakk. I tested a build on 1.73 and it worked like a charm - it didn't require me to bump the rust version.

**Comment by kivikakk** [comment]
Awesome, really glad! 😊  Thanks for testing it!

File: script/cibuild [hunk 1]
```diff
@@ -8,54 +8,42 @@ fi
 
 failed=0
 
-if [ x"$SPEC" = "xtrue" ]; then
```
Comment (digitalmoksha) [inline script/cibuild:11]: Nice cleanup, removing the `$SPEC`

**Comment by digitalmoksha** [comment]
@kivikakk so `propfuzz.rs` is no longer needed to do fuzzing?

**Comment by kivikakk** [comment]
That used https://github.com/facebookarchive/propfuzz, which (as the URL implies!) isn't actively worked on these days and never made it past an initial test release. (IIRC `propfuzz` predated `cargo-fuzz`, but yeah, we're in a much better position with the latter now.)

File: script/cibuild (hunk 1 above)
Comment (kivikakk) [reply script/cibuild:11]: Ta! It was always ugly; really old, too.

**Comment by gjtorikian** [comment]
Bleh, looks like the latest dependabot updates didn't merged because the `requires` MSRV tests failed. Maybe we should just chuck that automation—the longest time interval for the check is a month, which isn't going to really change anything in terms of support for MSRV (I would assume that would be on the order of months, if not years).

**Comment by digitalmoksha** [comment]
Ack, I think you might be right. I don't suppose dependabot can be made to have awareness of what our MSRV is and only recommend those updates?

What about just commenting it out for now, instead of ripping it out? Or do you think it's cleaner just to pull it.

**Comment by kivikakk** [comment]
> What about just commenting it out for now, instead of ripping it out? Or do you think it's cleaner just to pull it.

Always delete! It'll be there in revision history when we need it again.


> the longest time interval for the check is a month, which isn't going to really change anything in terms of support for MSRV (I would assume that would be on the order of months, if not years).

I'm inclined to agree. :/ Will remove for now. Thanks for all your work on that! I feel like we'll be restoring it in no time, the way the days seem to pass lately.

**Comment by gjtorikian** [comment]
> I don't suppose dependabot can be made to have awareness of what our MSRV is and only recommend those updates?

Afraid not. The same issue @kivikakk linked to in rust-core is what's holding up dependabot 🙃 https://github.com/dependabot/dependabot-core/issues/5423

**Pull request merged**